machine/v1beta1/types_awsplacementgroup.go

```diff
@@ -0,0 +1,164 @@
+)
+
+// AWSPlacementGroupType represents the valid values for the Placement GroupType field.
+type AWSPlacementGroupType string
+
+const (
+	// AWSClusterPlacementGroupType is the "Cluster" placement group type.
+	// Cluster placement groups place instances close together to improve network latency and throughput.
+	AWSClusterPlacementGroupType AWSPlacementGroupType = "Cluster"
+	// AWSPartitionPlacementGroupType is the "Partition" placement group type.
+	// Partition placement groups reduce the likelihood of hardware failures
+	// disrupting your application's availability.
+	// Partition placement groups are recommended for use with large scale
+	// distributed and replicated workloads.
+	AWSPartitionPlacementGroupType AWSPlacementGroupType = "Partition"
+	// AWSSpreadPlacementGroupType is the "Spread" placement group type.
+	// Spread placement groups place instances on distinct racks within the availability
+	// zone. This ensures instances each have their own networking and power source
+	// for maximum hardware fault tolerance.
+	// Spread placement groups are recommended for a small number of critical instances
+	// which must be kept separate from one another.
+	// Using a Spread placement group imposes a limit of seven instances within
+	// the placement group within a single availability zone.
+	AWSSpreadPlacementGroupType AWSPlacementGroupType = "Spread"
+)
+
+// AWSPlacementGroupUnion is a discriminated union of placement group configuration.
+// +union
+type AWSPlacementGroupUnion struct {
+	// GroupType specifies the type of AWS placement group to use for this Machine.
+	// This parameter is only used when a Machine is being created and the named
+	// placement group does not exist.
+	// Valid values are "Cluster", "Partition", "Spread".
+	// This value is required and, in case a placement group already exists, will be
+	// validated against the existing placement group.
+	// Note: If the value of this field is "Spread", Machines created within the group
+	// may no have placement.Tenancy set
+	// to "dedicated".
+	// +kubebuilder:validation:Enum:="Cluster";"Partition";"Spread"
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +required
+	GroupType AWSPlacementGroupType `json:"groupType,omitempty"`
+
+	// +optional
+	Partition *AWSPartitionPlacement `json:"partition,omitempty"`
+}
+
+// AWSPartitionPlacement defines the configuration for partition placement groups.
+type AWSPartitionPlacement struct {
+	// Count specifies the number of partitions for a Partition placement
+	// group. This value is only observed when creating a placement group and
+	// only when the `groupType` is set to `Partition`.
+	// Note the partition count of a placement group cannot be changed after creation.
+	// If unset, AWS will provide a default partition count.
+	// This default is currently 2.
+	// Note: When using more than 2 partitions, placement.tenancy "dedicated" on Machines
```

Comment:
small nit: Noticed `placement.tenancy` is lowercase here but uppercase within the `AWSPlacementGroupUnion` type. It'd be good to have them similar.

Reply:
perhaps we could make this a little clearer too, eg
```suggestion
	// Note: When using more than 2 partitions, the "dedicated" tenancy option on Machines
```